sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateRunDao.java

```diff
@@ -0,0 +1,386 @@
+    Criteria criteria = currentSession().createCriteria(RunImpl.class);
+    @SuppressWarnings("unchecked")
+    List<Run> records = criteria.list();
+    return records;
+  }
+
+  @Override
+  public int count() throws IOException {
+    Criteria criteria = currentSession().createCriteria(RunImpl.class);
+    return ((Long) criteria.setProjection(Projections.rowCount()).uniqueResult()).intValue();
+  }
+
+  @Override
+  public boolean remove(Run run) throws IOException {
+    if (run.isDeletable()) {
+      currentSession().delete(run);
+
+      Run testIfExists = get(run.getId());
+      return testIfExists == null;
+    } else {
+      return false;
+    }
+  }
+
+  @Override
+  public Run getLatestStartDateRunBySequencerPartitionContainerId(long containerId) throws IOException {
+    Criteria criteria = currentSession().createCriteria(RunImpl.class);
+    criteria.createAlias("containers", "spc");
+    criteria.add(Restrictions.eq("spc.id", containerId));
+    criteria.addOrder(Order.desc("status.startDate"));
+    criteria.setMaxResults(1);
+    return (Run) criteria.uniqueResult();
+  }
+
+  @Override
+  public Run getLatestRunIdRunBySequencerPartitionContainerId(long containerId) throws IOException {
+    Criteria criteria = currentSession().createCriteria(RunImpl.class);
+    criteria.createAlias("containers", "spc");
+    criteria.add(Restrictions.eq("spc.id", containerId));
+    criteria.addOrder(Order.desc("run.id"));
+    criteria.setMaxResults(1);
+    return (Run) criteria.uniqueResult();
+  }
+
+  private static final String[] searchProperties = new String[] { "name", "alias", "description" };
+
+  private Criterion searchRestrictions(String querystr) {
+    String str = DbUtils.convertStringToSearchQuery(querystr);
+
+    Criterion[] criteria = new Criterion[searchProperties.length];
+    for (int i = 0; i < searchProperties.length; i++) {
+      criteria[i] = Restrictions.ilike(searchProperties[i], str, MatchMode.ANYWHERE);
+    }
+    return Restrictions.or(criteria);
+  }
+
+  @Override
+  public List<Run> listBySearch(String query) throws IOException {
+    Criteria criteria = currentSession().createCriteria(RunImpl.class);
+    criteria.add(searchRestrictions(query));
+    @SuppressWarnings("unchecked")
+    List<Run> records = criteria.list();
+    return records;
+  }
+
+  @Override
+  public Run getByAlias(String alias) throws IOException {
+    Criteria criteria = currentSession().createCriteria(RunImpl.class);
+    criteria.add(Restrictions.eq("alias", alias));
+    @SuppressWarnings("unchecked")
+    List<Run> records = criteria.list();
+    if (records.size() == 1) {
+      return records.iterator().next();
+    } else {
+      throw new IOException("Found more than one run by this name");
+    }
+  }
+
+  @Override
+  @Deprecated
+  public List<Run> listByExperimentId(long experimentId) throws IOException {
+    return Collections.emptyList();
+  }
+
+  // TODO: confirm that this works as intended
```

Comment:
Make sure the methods marked with this have decent tests and then remove the TODO tags. Tests should fail if it doesn't work as intended